avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/JacksonAvroParserImpl.java

```diff
@@ -628,6 +628,9 @@ private final String _finishShortText(int len) throws IOException
         final int[] codes = sUtf8UnitLengths;
         do {
             i = inputBuf[inPtr++] & 0xFF;
+            if (inPtr >= inputBuf.length) {
```

Comment:
Not quite correct: need to compare to `end`, not physical buffer end (two may or may not be same).

But I can work with this, thanks!